packages/backend/src/services/PromoteService/PromoteService.ts

```diff
@@ -860,9 +860,10 @@ export class PromoteService extends BaseService {
             .filter((change) => change.action === PromotionAction.CREATE)
             .map(async (spaceChange) => {
                 const { data: spaceChangeFromBase } = spaceChange;
-                const isNestedSpace = await this.spaceModel.isRootSpace(
+                const isRootSpace = await this.spaceModel.isRootSpace(
```

Comment:
🤦 

good spot. I just tested this with and without this change. 
confirmed that with your fix the hierarchy is built correctly.

Reply:
Tested also with one root being already in the upstream project. and saw the child and grandchild being promoted in the upstream